mrao2/newtowndogs
# updated sql statements to delete blogs with comments

was getting this message when trying to delete a blog with a comment attached to it:
![image](https://github.com/mrao2/newtowndogs/assets/118093209/8be8f157-e583-4de7-acfa-c67a18f2e82c)
Error: ER_ROW_IS_REFERENCED_2: Cannot delete or update a parent row: a foreign key constraint fails (`newtowndogs`.`comments`, CONSTRAINT `comments_ibfk_2` FOREIGN KEY (`BlogId`) REFERENCES `blogs` (`BlogId`))
    at Sequence._packetToError (C:\Users\paule\Projects\launchcode\Project\newtowndogs\backend\node_modules\mysql\lib\protocol\sequences\Sequence.js:47:14)
    at Query.ErrorPacket (C:\Users\paule\Projects\launchcode\Project\newtowndogs\backend\node_modules\mysql\lib\protocol\sequences\Query.js:79:18)
    at Protocol._parsePacket (C:\Users\paule\Projects\launchcode\Project\newtowndogs\backend\node_modules\mysql\lib\protocol\Protocol.js:291:23)
    at Parser._parsePacket (C:\Users\paule\Projects\launchcode\Project\newtowndogs\backend\node_modules\mysql\lib\protocol\Parser.js:433:10)
    at Parser.write (C:\Users\paule\Projects\launchcode\Project\newtowndogs\backend\node_modules\mysql\lib\protocol\Parser.js:43:10)
    at Protocol.write (C:\Users\paule\Projects\launchcode\Project\newtowndogs\backend\node_modules\mysql\lib\protocol\Protocol.js:38:16)
    at Socket.<anonymous> (C:\Users\paule\Projects\launchcode\Project\newtowndogs\backend\node_modules\mysql\lib\Connection.js:88:28)
    at Socket.<anonymous> (C:\Users\paule\Projects\launchcode\Project\newtowndogs\backend\node_modules\mysql\lib\Connection.js:526:10)
    at Socket.emit (node:events:513:28)
    at addChunk (node:internal/streams/readable:324:12)
    --------------------
    at Protocol._enqueue (C:\Users\paule\Projects\launchcode\Project\newtowndogs\backend\node_modules\mysql\lib\protocol\Protocol.js:144:48)
    at Connection.query (C:\Users\paule\Projects\launchcode\Project\newtowndogs\backend\node_modules\mysql\lib\Connection.js:198:25)
    at Delete (C:\Users\paule\Projects\launchcode\Project\newtowndogs\backend\SqlFunctions\Delete.js:12:20)
:284:15
    at param (C:\Users\paule\Projects\launchcode\Project\newtowndogs\backend\node_modules\express\lib\router\index.js:365:14) {
  code: 'ER_ROW_IS_REFERENCED_2',
  errno: 1451,
  sqlMessage: 'Cannot delete or update a parent row: a foreign key constraint fails (`newtowndogs`.`comments`, CONSTRAINT `comments_ibfk_2` FOREIGN KEY (`BlogId`) REFERENCES `blogs` (`BlogId`))',
  sqlState: '23000',
  index: 0,
  sql: "DELETE FROM blogs WHERE BlogId = '1'"
}

Node.js v18.16.1
[nodemon] app crashed - waiting for file changes before starting...

Updated the sql statements in two places. 
1. blogs_sql-- added an index reference to blog id on line 11 for safety:    INDEX (BlogId),
2. comments.sql --added ON DELETE CASCADE on line 10 to avoid a constraint error when trying to delete the blog with a comment attached. 

Testing steps:
1. In mysql drop table comments
2. drop table blogs
3. run new create statement for blogs: 
CREATE TABLE blogs (
  BlogId INT AUTO_INCREMENT PRIMARY KEY,
  Author VARCHAR(255),
  Title VARCHAR(255),
  Body VARCHAR(255),
  Likes VARCHAR(255),
  Category VARCHAR(255),
  Comments VARCHAR(255),
  username VARCHAR (45),
   INDEX (username),
   INDEX (BlogId),
  FOREIGN KEY (username) REFERENCES login_app(username)
);
4. run new create statement for comments:
CREATE TABLE comments (
  CommentId INT AUTO_INCREMENT PRIMARY KEY,
  BlogId INT,
  Comment_Author VARCHAR(255),
  Comment_Body VARCHAR(255),
  username VARCHAR (45),
   INDEX (username),
   INDEX (BlogId),
  FOREIGN KEY (username) REFERENCES login_app(username),
  FOREIGN KEY (BlogId) REFERENCES blogs(BlogId) ON DELETE CASCADE
);

5. run the app, create a new blog, add comments to the blog, then delete the entire blog.
 ![image](https://github.com/mrao2/newtowndogs/assets/118093209/bace80f2-933a-4af2-a8be-78035e008f9a)
6. Should see the All Blogs page loaded after:

--also made some homepage updates and navbar updates
![image](https://github.com/mrao2/newtowndogs/assets/118093209/3331c5cd-4492-44ad-a087-547d1fb39246)
